src/tools/crossgen2/Common/TypeSystem/Sorting/TypeSystemComparer.cs

```diff
@@ -24,7 +27,7 @@ namespace Internal.TypeSystem
     // to sort itself with respect to other instances of the same type.
     // Comparisons between different categories of types are centralized to a single location that
     // can provide rules to sort them.
-    public class TypeSystemComparer
+    public class TypeSystemComparer : IComparer<MethodDesc>
```

Comment:
Why did you need to add the `IComparer<MethodDesc>` here?

Reply:
We need it when sorting the MethodDescs in ReadyToRunTableManager.

Reply:
This seems odd to me. Why do we need to implement `IComparer<MethodDesc>` only, and not do that for types/fields or other type system entities?

Reply:
OrderBy needs an IComparer<MethodDesc> (and so would sort).

Reply:
Why does `_methodsGenerated` have to be a Hashset? if we make it into a List<MethodDesc>, we can easily write it without any of the Linq stuff:
``` c#
var comparer = new TypeSystemComparer();
_methodsGenerated.Sort((x, y) => comparer.Compare(x, y));
```

Then you won't have to add the IComparer interface here. You'll just need some check that sorting has been done to catch any additions to the list after the sorting.

It just looks odd that we're adding this interface for the purpose of using Linq sorting just for methods, and we're not adding the interface for the other compare functionalities.

cc @nattress @MichalStrehovsky for opinions